### ISSUE TITLE
Modal: Screanreader doesn't announce nested modal

### ISSUE THREAD

**Issue by nikitaorliak-cengage** [cengage/react-magma#1761]
**Component**
Modal 

**Description of bug**
Screanreader doesn't announce nested modal

**Steps to reproduce**
- Modal on the docs site;
- `Nested Modals` section;
- Turn On VoiceOver;
- Open modal -> Open nested modal
- Notice that screenreader doesn't announce header in the nested modal
![image](https://github.com/user-attachments/assets/55e37ee1-6abd-42ea-b9b6-b9a2d9e5b1ba)

**Note**
- If you focus on "Close button" - All modal will be announced;
- Confirmation modal in the `Close Modal With Confirmation` section works fine, screanreader announce it.
